Sources/StreamChatUI/Utils/DateFormatter+Extensions.swift

```diff
@@ -5,33 +5,41 @@
 import Foundation
 
 extension DateFormatter {
-    static func makeDefault() -> DateFormatter {
+    /// A formatter that converts the message timestamp to textual representation.
+    public static var messageTimestamp: DateFormatter = {
         let formatter = DateFormatter()
         formatter.dateStyle = .none
         formatter.timeStyle = .short
         formatter.locale = Locale.autoupdatingCurrent
         return formatter
-    }
+    }()
     
-    /// Formatter that is used to format date for scrolling overlay that should display
-    /// day when message below was sent
-    public static let messageListDateOverlay: DateFormatter = {
-        let df = DateFormatter()
-        df.setLocalizedDateFormatFromTemplate("MMMdd")
-        df.locale = .autoupdatingCurrent
-        return df
+    /// A formatter that converts the message date to textual representation and used as a top overlay in the message list.
+    public static var messageListDateOverlay: DateFormatter = {
```

Comment:
The approach we'd like to follow is:
```swift
struct Appearance {
  var dateFormatters = DateFormatters()

  struct DateFormatters {
     var formatter1 = …
     var formatter2 = …
  }
}
```

The **messageListDateOverlay** is the only publicly exposed static formatter (which we have to maintain) that is in conflict with the `Appearance`, right? 🤔 What if we make it forwarding to `Appearance.default`:
```swift
extension DateFormatter {
    @available(*, deprecated, message: "Please use `Appearance.default.dateFormatters.formatter1` instead")
    public static var messageListDateOverlay: DateFormatter {
      get { Appearance.default.dateFormatters.formatter1 }
      set { Appearance.default.dateFormatters.formatter1 = newValue }
    }
}
```

It's fine formatters are not static thanks to `Appearance.default` used in 99% of cases. A shared instance get instantiated with all its props just once when first accessed. And it's still possible to customize the formatter for the specific view-hierarchy subtree.
 
@nuno-vieira  WDYT?

Reply:
we can also make dateFormatters and all underlying pros `lazy`

Reply:
We also have Number formatters. I will change the API and try to change to `Appearance.default.formatters` 👍

Reply:
✅